test/e2e/csv_e2e_test.go

```diff
@@ -4474,47 +4465,65 @@ func awaitCSV(c versioned.Interface, namespace, name string, checker csvConditio
 }
 
 func waitForDeployment(namespace string, c operatorclient.ClientInterface, name string) error {
-	return wait.Poll(pollInterval, pollDuration, func() (bool, error) {
-		_, err := c.GetDeployment(namespace, name)
+	var err error
+
+	Eventually(func() (bool, error) {
+		_, err = c.GetDeployment(namespace, name)
 		if err != nil {
-			if apierrors.IsNotFound(err) {
```

Comment:
I'm not sure it's safe to remove this?

Reply:
The reasoning is that regardless of the error ("not found" or "i/o error"), we have to continue to retry until the deployment is found. Note that this has changed from `wait.Poll()` to `Eventually()` which have ever-so-slightly different semantics.
* `Eventually()` continues until the first return value meets the test (`Should(BeTrue())`) and all subsequent return values are "0".
* `wait.Poll()` continues until the `bool` return value is `true` or the `error` return value is `non-nil`. In this case, the `wait.Poll()` was terminating on any error other than "not found", including "i/o errors", which should be retried.

Reply:
(Also, this is a WIP, so this is not yet final. 😄)

Reply:
Do we really suffer from failing to reach the API server during tests? Yikes.